scripts/agent_step_zkai_stwo_composition.py

```
@@ -0,0 +1,617 @@
+import copy
+import csv
+import hashlib
+import importlib.util
+import io
+import json
+import os
+import pathlib
+import shutil
+import subprocess
+import sys
+from typing import Any, Callable
```

Ruff (0.15.12)

[warning] Import from `collections.abc` instead: `Callable`

Import from `collections.abc`

(UP035)